src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -1094,6 +1210,15 @@ export default class RouteModel extends BaseModel {
         return amendedStarLeg;
     }
 
+    /**
+     * Return a STAR leg based on the provided leg, except with the new specified exit
+     *
+     * @for RouteModel
+     * @method _createAmendedStarLegUsingDifferentExitName
+     * @param exitFixName {string} name of STAR exit to use for the new STAR leg
+     * @param legIndex {number} index of leg in the #_legCollection
+     * @return {LegModel}
+     */
     _createAmendedStarLegUsingDifferentExitName(exitFixName, legIndex) {
         const divergentLegModel = this._legCollection[legIndex];
```

Comment:
is there ever a case where `divergentLegModel` would be `undefined`?

Reply:
No, we will only be able to get here if the routes have continuity, which means _some_ leg is the one where the routes diverge. So we should be well-covered on that front.